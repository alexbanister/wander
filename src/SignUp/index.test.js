import { shallow } from 'enzyme';
import { SignUp } from './index';
import React from 'react';


describe('SignUp', () => {
  it('should always match the snapshot', () => {
    const wrapper = shallow(<SignUp />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should change state', () => {
    const expected = {
      name: 'Billy Bob',
      email: 'email@email.com',
      password: 'password',
      retypePassword: 'password',
      signUpError: false,
      disabled: false,
      passwordValidationError: false
    };

    const wrapper = shallow(<SignUp
      handleSubmit={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();

    const name = wrapper.find('[type="text"]');
    const email = wrapper.find('[type="email"]');
    const password = wrapper.find('[type="password"]').first();
    const repassword = wrapper.find('[type="password"]').last();

    name.simulate('change', {target: {value: 'Billy Bob'}});
    email.simulate('change', {target: {value: 'email@email.com'}});
    password.simulate('change', {target: {value: 'password'}});
    repassword.simulate('change', {target: {value: 'password'}});

    expect(wrapper.state()).toEqual(expected);
  });
});
