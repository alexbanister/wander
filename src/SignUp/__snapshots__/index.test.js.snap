// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    onChange={[Function]}
                    placeholder="Name"
                    type="text"
          />
          <input
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
          />
          <input
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
          />
          <input
                    onChange={[Function]}
                    placeholder="Please Retype Password"
                    type="password"
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Sign Up
          </button>
</form>,
        "Already have an account?",
        <br />,
        <Link
          replace={false}
          to="/login"
>
          Log In
</Link>,
      ],
      "className": "signup",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="Name"
              type="text"
/>,
            <input
              onChange={[Function]}
              placeholder="Email"
              type="email"
/>,
            <input
              onChange={[Function]}
              placeholder="Password"
              type="password"
/>,
            <input
              onChange={[Function]}
              placeholder="Please Retype Password"
              type="password"
/>,
            <button
              disabled={true}
              type="submit"
>
              Sign Up
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Email",
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Please Retype Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "Already have an account?",
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Log In",
          "replace": false,
          "to": "/login",
        },
        "ref": null,
        "rendered": "Log In",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        onChange={[Function]}
                        placeholder="Name"
                        type="text"
            />
            <input
                        onChange={[Function]}
                        placeholder="Email"
                        type="email"
            />
            <input
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
            />
            <input
                        onChange={[Function]}
                        placeholder="Please Retype Password"
                        type="password"
            />
            <button
                        disabled={true}
                        type="submit"
            >
                        Sign Up
            </button>
</form>,
          "Already have an account?",
          <br />,
          <Link
            replace={false}
            to="/login"
>
            Log In
</Link>,
        ],
        "className": "signup",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder="Name"
                type="text"
/>,
              <input
                onChange={[Function]}
                placeholder="Email"
                type="email"
/>,
              <input
                onChange={[Function]}
                placeholder="Password"
                type="password"
/>,
              <input
                onChange={[Function]}
                placeholder="Please Retype Password"
                type="password"
/>,
              <button
                disabled={true}
                type="submit"
>
                Sign Up
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Email",
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Please Retype Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "button",
            },
          ],
          "type": "form",
        },
        "Already have an account?",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Log In",
            "replace": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": "Log In",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SignUp should change state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    handleSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    onChange={[Function]}
                    placeholder="Name"
                    type="text"
          />
          <input
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
          />
          <input
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
          />
          <input
                    onChange={[Function]}
                    placeholder="Please Retype Password"
                    type="password"
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Sign Up
          </button>
</form>,
        "Already have an account?",
        <br />,
        <Link
          replace={false}
          to="/login"
>
          Log In
</Link>,
      ],
      "className": "signup",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="Name"
              type="text"
/>,
            <input
              onChange={[Function]}
              placeholder="Email"
              type="email"
/>,
            <input
              onChange={[Function]}
              placeholder="Password"
              type="password"
/>,
            <input
              onChange={[Function]}
              placeholder="Please Retype Password"
              type="password"
/>,
            <button
              disabled={true}
              type="submit"
>
              Sign Up
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Email",
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Please Retype Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "Already have an account?",
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Log In",
          "replace": false,
          "to": "/login",
        },
        "ref": null,
        "rendered": "Log In",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        onChange={[Function]}
                        placeholder="Name"
                        type="text"
            />
            <input
                        onChange={[Function]}
                        placeholder="Email"
                        type="email"
            />
            <input
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
            />
            <input
                        onChange={[Function]}
                        placeholder="Please Retype Password"
                        type="password"
            />
            <button
                        disabled={true}
                        type="submit"
            >
                        Sign Up
            </button>
</form>,
          "Already have an account?",
          <br />,
          <Link
            replace={false}
            to="/login"
>
            Log In
</Link>,
        ],
        "className": "signup",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder="Name"
                type="text"
/>,
              <input
                onChange={[Function]}
                placeholder="Email"
                type="email"
/>,
              <input
                onChange={[Function]}
                placeholder="Password"
                type="password"
/>,
              <input
                onChange={[Function]}
                placeholder="Please Retype Password"
                type="password"
/>,
              <button
                disabled={true}
                type="submit"
>
                Sign Up
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Email",
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Please Retype Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "button",
            },
          ],
          "type": "form",
        },
        "Already have an account?",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Log In",
            "replace": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": "Log In",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
